### PR TITLE
[DOCS] Expand information on using a runtime field without a script

### PR DESCRIPTION
**Pull request by lockewritesdocs**
To retrieve field values from `_source` with a runtime field, you can define a runtime field without a script, or prefix a field with `params._source`. This change expands on the existing section for defining a runtime field without a script to provide additional guidance, and also advocate for using `doc_values` when available. 

Preview link: https://elasticsearch_73219.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/runtime-mapping-fields.html#runtime-fields-scriptless

### CODE REVIEW

File: docs/reference/mapping/runtime.asciidoc [hunk 1]
```diff
@@ -145,10 +145,34 @@ PUT my-index
...
+Defining a runtime field in the mapping definition without a script is the
+recommended option. At query time, {es} implicitly looks in `_source` for a
+field with the same name as the runtime field and returns a value if one exists.
+If a field with the same name doesn’t exist, the response doesn't include any
+values for that runtime field.
+
+Additionally, a runtime field without a script ignores errors as if <<ignore-malformed,`ignore_malformed`>> is `true`. Your runtime field will
+return values for the specified field, regardless of how the data is formatted.
+This behavior is really useful for exploring your data and determining how you
+might want to <<runtime-mapping-fields,map a runtime field>>.
```
Comment (javanna) [inline docs/reference/mapping/runtime.asciidoc:172]: I am not super sure we want to promote this, especially knowing that we want to change it, see https://github.com/elastic/elasticsearch/issues/72143

**Review by javanna — COMMENTED** [review]
left a couple of comments

File: docs/reference/mapping/runtime.asciidoc (hunk 1 above)
Comment (lockewritesdocs) [reply docs/reference/mapping/runtime.asciidoc:172]: Thanks for identifying the related issue. I'll remove this section.

File: docs/reference/mapping/runtime.asciidoc [hunk 2]
```diff
@@ -145,10 +145,34 @@ PUT my-index
...
+available by default, or where you choose to not index specific fields or set
+`doc_values` to `false`. In these instances, you have two options to retrieve
+values directly from `_source` using runtime fields:
+
+* Define a runtime field in the mapping without a script (*recommended*)
+* Prefix the field you want to retrieve values for with `params._source` (such
+as `params._source.day_of_week`)
+
+Defining a runtime field in the mapping definition without a script is the
+recommended option. At query time, {es} implicitly looks in `_source` for a
```
[the remark this answers is not in the document]
Comment (lockewritesdocs) [reply docs/reference/mapping/runtime.asciidoc]: @javanna, I reworked this section to indicate that users typically want to use a runtime field for manipulating data in some way, and then explained when they might want to define a runtime field without a script. Let me know what you think of the rewrite, and if it addresses your comments.

File: docs/reference/mapping/runtime.asciidoc [hunk 3]
```diff
@@ -145,34 +145,10 @@ PUT my-index
...
-Additionally, a runtime field without a script ignores errors as if <<ignore-malformed,`ignore_malformed`>> is `true`. Your runtime field will
-return values for the specified field, regardless of how the data is formatted.
-This behavior is really useful for exploring your data and determining how you
-might want to <<runtime-mapping-fields,map a runtime field>>.
-
-The following request adds a `day_of_week` runtime field to `my-index` without
-a script. 
+Runtime fields typically include a Painless script that manipulates data in some
+way. However, there are instances where you might define a runtime field
+_without_ a script. For example, if you want to retrieve a single value from `_source` without making changes, you don't need a script. You can just create
```
Comment (javanna) [inline docs/reference/mapping/runtime.asciidoc:150]: nit: single value can be misleading. A single field can hold multiple values :)

File: docs/reference/mapping/runtime.asciidoc [hunk 4]
```diff
@@ -188,6 +164,26 @@ PUT my-index/
...
+exists. If a field with the same name doesn’t exist, the response doesn't
+include any values for that runtime field.
+
+In most cases, you should retrieve field values through
+<<doc-values,`doc_values`>> whenever possible. Accessing `doc_values` with a
+runtime field is faster than retrieving values from `_source` because of how
+data is loaded from Lucene.
+
+However, there are cases such as `text` fields where `doc_values` are not
+available by default, or where you choose to not index specific fields or set
```
Comment (javanna) [inline docs/reference/mapping/runtime.asciidoc:178]: or where you choose to disable `doc_values`. Indexing here can be confusing: you may have a mapped field with index:true and doc_value: false. That field is not indexed and you need to load it from _source if you want to access its values from a script. Yet the field is searchable, hence you don't need a runtime field to be able to query it, but it is not aggregatable.

File: docs/reference/mapping/runtime.asciidoc [hunk 5]
```diff
@@ -188,6 +164,26 @@ PUT my-index/
...
+`doc_values` to `false`. In these instances, define a runtime field without a
+script to retrieve values directly from `_source`.
+
+NOTE: You can alternatively prefix the field you want to retrieve values for
+with `params._source` (such as `params._source.day_of_week`). For simplicity,
+defining a runtime field in the mapping definition without a script is the
+recommended option.
```
Comment (javanna) [inline docs/reference/mapping/runtime.asciidoc:185]: add "when possible" at the end?

**Review by javanna — COMMENTED** [review]
much better, thanks, left a couple more nits.

File: docs/reference/mapping/runtime.asciidoc (hunk 4 above)
Comment (lockewritesdocs) [reply docs/reference/mapping/runtime.asciidoc]: Thanks @javanna, I'll clarify that use case.

File: docs/reference/mapping/runtime.asciidoc [hunk 6]
```diff
@@ -164,6 +164,32 @@ PUT my-index/
...
+
+In most cases, retrieve field values through
+<<doc-values,`doc_values`>> whenever possible. Accessing `doc_values` with a
+runtime field is faster than retrieving values from `_source` because of how
+data is loaded from Lucene.
+
+However, there are cases where retrieving fields from `_source` is necessary.
+For example, `text` fields do not have `doc_values` available by default, so you
+have to retrieve values from `_source`. In other instances, you might choose to
+not index specific fields or choose to disable `doc_values` on an index.
```
Comment (javanna) [inline docs/reference/mapping/runtime.asciidoc:180]: the fact that you index specific fields is irrelevant as to whether you can load its values using `doc` from a script. Doc_values are required for that. Shall we shorten this sentence to "you might choose to disable doc_value on a field ?

File: docs/reference/mapping/runtime.asciidoc [hunk 7]
```diff
@@ -164,6 +164,32 @@ PUT my-index/
...
+
+With `doc_values` disabled, you could have a field in your index mapping where
+`index` is `true` and `doc_values` is `false`. This combination can be
+confusing because although the field isn't indexed, it is _searchable_. In this
+case, define a runtime field without a script to retrieve values directly
+from `_source`.
```
Comment (javanna) [inline docs/reference/mapping/runtime.asciidoc:186]: I find that this sentence, while it's trying to clarify things, adds even more confusion :) I would remove it.

**Review by javanna — APPROVED** [review]
left two comments, LGTM otherwise

**Pull request merged**